go/pkg/metastore/db/dao/segment.go

```diff
@@ -51,11 +51,16 @@ func (s *segmentDb) Insert(in *dbmodel.Segment) error {
 }
 
 func (s *segmentDb) GetSegments(id types.UniqueID, segmentType *string, scope *string, collectionID types.UniqueID) ([]*dbmodel.SegmentAndMetadata, error) {
+	if collectionID == types.NilUniqueID() {
+		return nil, common.ErrMissingCollectionID
+	}
+
 	var segments []*dbmodel.SegmentAndMetadata
 
 	query := s.db.Table("segments").
 		Select("segments.id, segments.collection_id, segments.type, segments.scope, segments.file_paths, segment_metadata.key, segment_metadata.str_value, segment_metadata.int_value, segment_metadata.float_value, segment_metadata.bool_value").
 		Joins("LEFT JOIN segment_metadata ON segments.id = segment_metadata.segment_id").
+		Where("segments.collection_id = ?", collectionID.String()).
```

Comment:
Will this affect the query performance of get segments? (I don't recall the schema but am guessing no)

Reply:
yes, it should massively speed it up--this was the whole point of this PR haha
because the index is composite over `(collection_id, id)` on segments, querying only by the segment ID won't use the index

Reply:
ah yes

Reply:
lol